### PR TITLE
Auto-Contribute list percentages now do not underflow

### PR DESCRIPTION
**Pull request by jasonrsadler**
Fixes brave/brave-browser#3668

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:

1. Start Brave clean profile, enable rewards.
2. Add a lot of sites to auto-contribute
3. Go to Rewards page and start excluding sites.
4. Make sure highest percentage does not go above 100%

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers_unittest.cc [hunk 1]
```diff
@@ -3,6 +3,9 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "base/logging.h"
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers_unittest.cc]: is this one needed?

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc [hunk 2]
```diff
@@ -461,10 +461,14 @@ void BatPublishers::synopsisNormalizerInternal(
     }
     if (percents.size() != 0) {
       if (totalPercents > 100) {
-        percents[valueToChange] -= 1;
+        if (percents[valueToChange] != 0) {
+          percents[valueToChange] -= 1;
+        }
         totalPercents -= 1;
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: this one needs to go into if as well, as we didn't actually reduce total % if `percents[valueToChange]` is 0

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc [hunk 3]
```diff
@@ -461,10 +461,14 @@ void BatPublishers::synopsisNormalizerInternal(
...
       } else {
-        percents[valueToChange] += 1;
+        if (percents[valueToChange] != 100) {
+          percents[valueToChange] += 1;
+        }
         totalPercents += 1;
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: same as above

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers_unittest.cc (hunk 1 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers_unittest.cc]: Nope removed

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc (hunk 2 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: Good catch. Fixed.

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc (hunk 3 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_publishers.cc]: Fixed.

**Pull request merged**